privilege/privileges/cache.go

```diff
@@ -212,9 +212,21 @@ func compareHost(x, y string) int {
 		return -1
 	}
 
-	if strings.HasSuffix(y, `%`) {
-		if !strings.HasSuffix(x, `%`) {
+	// One of them end with `%`.
+	xEnd := strings.HasSuffix(x, `%`)
+	yEnd := strings.HasSuffix(y, `%`)
+	if xEnd || yEnd {
+		switch {
+		case !xEnd && yEnd:
 			return -1
+		case xEnd && !yEnd:
+			return 1
+		case xEnd && yEnd:
+			// 192.168.199.% smaller than 192.168.%
```

Comment:
what about '192.%.128.%' and '192.168.1.%' ?
`create user 'test'@'192.%.1.%'` will be ok in MySQL.

Reply:
The result will be nondeterministic

Reply:
OK
We may add a FIXME label here.

Reply:
I add a TODO in at the beginning of this function.